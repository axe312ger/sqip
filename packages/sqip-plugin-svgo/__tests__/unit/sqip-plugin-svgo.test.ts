import SVGO from 'svgo'
import SvgoPlugin from '../../src/sqip-plugin-svgo'

jest.mock('svgo')

const optimizeMock = jest.fn(() => Promise.resolve({ data: 'mocked' }))

SVGO.mockImplementation(function SVGOmock() {
  this.optimize = optimizeMock
})

test('runSVGO', async () => {
  const svgoPlugin = new SvgoPlugin({})
  const inputSVG = Buffer.from('<svg />')
  await svgoPlugin.apply(inputSVG)
  expect(SVGO).toHaveBeenCalledTimes(1)
  expect(SVGO.mock.calls[0]).toMatchSnapshot()
  expect(optimizeMock).toHaveBeenCalledTimes(1)
  expect(optimizeMock.mock.calls[0][0]).toBe(inputSVG)
})
