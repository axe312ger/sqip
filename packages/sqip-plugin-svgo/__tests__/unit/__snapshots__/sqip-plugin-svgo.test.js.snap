// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`runSVGO 1`] = `
Array [
  Object {
    "floatPrecision": 1,
    "multipass": true,
    "plugins": Array [
      Object {
        "removeViewBox": false,
      },
    ],
  },
]
`;
